### DISCUSSION TITLE
Parameter mismatch between hetere_conv->forward and gcn_conv->forward?

### DISCUSSION THREAD

**Discussion by soesto** [pyg-team/pytorch_geometric#5483] [General]
I am running into the following (probably user error) but this is as it goes: (using 2.1.0)

1) Created a heterogeneous convolution wrapper like in the docs -> https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#trainfunc but with only one GCNConv per layer (2 layers) and one linear (not relevant at this point)

2) When my wrapper's forward gets invoked it invokes HeteroConv->forward as expected but then (in line 130 of hetero_conv.py) it invokes gcn_conv.py->forward with a tuple. gcn_conv.py takes a tensor. This causes a runtime error.

What am I missing?

Thanks a lot!

**Comment by soesto** [comment]
Code that repoduces the problem.
```
class HeterogenousGNN(torch.nn.Module):
    def __init__(self, hidden_channels: int, out_channels: int, in_channels: int = -1):
        super().__init__()
        required_convs = {}
        required_convs[("X", "Y", "Z")] = GCNConv(in_channels, hidden_channels, add_self_loops=False) 

        self.convs = torch.nn.ModuleList()
        for i in range(0, 1):
            hc = HeteroConv(required_convs, aggr='sum')
            self.convs.append(hc)
        self.lin = Linear(hidden_channels, out_channels)

    def forward(self, x_dict, edge_index_dict) :
        for conv in self.convs:
            x_dict = conv(x_dict, edge_index_dict) # Error happens here
            x_dict = {key: x.relu() for key, x in x_dict.items()}
        return self.lin(x_dict['Z'])`
```

**Comment by EdisonLeeeee** [comment]
Currently `GCNConv` is not supported for bipartite graphs. You might need to used `SAGEConv` or other similar instead.

**Comment by soesto** [comment]
The graph is not bipartite, I just happened to pass only edges where dst != src. After reading some of the Kipf GCN whitepaper, it makes sense. Thanks!
